### PR TITLE
AMP-57987 Node Ampli JS/TS V2 examples

### PR DESCRIPTION
**Pull request by falconandy**

### CODE REVIEW

File: node/typescript/v2/AmpliApp/src/ampli/index.ts [hunk 1]
```diff
@@ -0,0 +1,906 @@
...
+  constructor(
+    public event_properties: EventWithTemplatePropertiesProperties,
+  ) {
+    this.event_properties = event_properties;
+  }
+}
+
+export type PromiseResult<T> = { promise: Promise<T | void> };
+
+const getVoidPromiseResult = () => ({ promise: Promise.resolve() });
```
Comment (justin-fiedler) [inline node/typescript/v2/AmpliApp/src/ampli/index.ts:513]: Can you please move this to the core SDK?

We could add PromiseReturn to https://github.com/amplitude/Amplitude-TypeScript/blob/main/packages/analytics-types/src/amplitude-promise.ts
And getVoidPromiseReturn to https://github.com/amplitude/Amplitude-TypeScript/blob/7d11fdbee2ece65b64cec873d2f2ab74ed5fecf3/packages/analytics-core/src/utils/return-wrapper.ts

Comment (falconandy) [reply node/typescript/v2/AmpliApp/src/ampli/index.ts:513]: That types are not used in the core SDK. Should we move them?

File: .github/workflows/ci-build-js-ts-test.yml [hunk 2]
```diff
@@ -7,8 +7,10 @@ on:
       - 'browser/typescript/v1/**'
       - 'browser/javascript/v2/**'
       - 'browser/typescript/v2/**'
-      - 'node/javascript/**'
```
[the remark this answers is not in the document]
Comment (falconandy) [reply .github/workflows/ci-build-js-ts-test.yml:10]: Done

File: node/typescript/v2/AmpliApp/src/ampli/index.ts (hunk 1 above)
Comment (justin-fiedler) [reply node/typescript/v2/AmpliApp/src/ampli/index.ts:513]: Hmm good question. I guess for now let's leave it in codegen.

File: node/javascript/v2/AmpliApp/src/ampli/index.js [hunk 3]
```diff
@@ -0,0 +1,642 @@
...
+  setGroup(userId, groupType, groupName, options) {
+    if (!this.isInitializedAndEnabled()) {
+      return getVoidPromiseResult();
+    }
+
+    if (userId) {
+      options = {...options,  user_id: userId};
+    }
+
+    // TODO: use options.
```
[the remark this answers is not in the document]
Comment (falconandy) [reply node/javascript/v2/AmpliApp/src/ampli/index.js]: Done.

Possible issues. It looks like:
1. the core SDK doesn't check if an event has non-empty either `user_id` or `device_id`
2. backend rejects all events in a batch if one of the events is invalid (for example, both `user_id` and `device_id` are empty)

**Review by justin-fiedler — APPROVED** [review]
Thank @falconandy looks great. Waiting to merge until other changes (ff, web, backend) are deployed.

**Pull request merged**